core/src/main/java/org/apache/calcite/rel/metadata/RelMdExpressionLineage.java

```diff
@@ -493,9 +493,8 @@ private static void createExpressions(RexBuilder rexBuilder,
       Map<RexInputRef, RexNode> singleMapping, Set<RexNode> result) {
     if (mapping.isEmpty()) {
       final RexReplacer replacer = new RexReplacer(singleMapping);
-      final List<RexNode> updatedPreds = new ArrayList<>(
-          RelOptUtil.conjunctions(
-              rexBuilder.copy(expr)));
+      final List<RexNode> updatedPreds = new ArrayList<>();
+      updatedPreds.add(rexBuilder.copy(expr));
```

Comment:
Micro optim, probably insignificant but since the list is always gonna have one element so let's keep it like that:

```suggestion
      final List<RexNode> updatedPreds = new ArrayList<>(1);
      updatedPreds.add(rexBuilder.copy(expr));
```